src/containers/Dashboard/DashboardContainer.tsx

```diff
@@ -13,14 +13,23 @@ import {useSelector} from "react-redux";
 import {RootState} from "../../app/rootReducer";
 import {useHistory} from "react-router-dom";
 import {Routes} from "../../constants/routes";
+import jwt_decode from "jwt-decode";
 
 export const DashboardContainer = (): ReactElement => {
     const [areElementsHidden, setElementsHidden] = useState(false);
     const history = useHistory();
     const stateToken = useSelector((state: RootState) => state.user.token);
 
+    const checkTokenExpireTime = (token: string | null): boolean => {
```

Comment:
This is kinda common for all pages in the app (even though we have dashboard now only) so would be good if you put it behind in the service, redux state logic. For example, you could move this function when initializing state with token. If it is expired set token in state as null. Then the authenticated routes logic that you wrote works automatically.